component/src/main/java/io/siddhi/extension/io/cdc/source/listening/RdbmsChangeDataCapture.java

```diff
@@ -0,0 +1,129 @@
+        } catch (NullPointerException | DataException ex) {
+            return detailsMap;
+        }
+        //match the change data's operation with user specifying operation and proceed.
+        if (operation.equalsIgnoreCase(CDCSourceConstants.INSERT) &&
+                op.equals(CDCSourceConstants.CONNECT_RECORD_INSERT_OPERATION)
+                || operation.equalsIgnoreCase(CDCSourceConstants.DELETE) &&
+                op.equals(CDCSourceConstants.CONNECT_RECORD_DELETE_OPERATION)
+                || operation.equalsIgnoreCase(CDCSourceConstants.UPDATE) &&
+                op.equals(CDCSourceConstants.CONNECT_RECORD_UPDATE_OPERATION)) {
+            Struct rawDetails;
+            List<Field> fields;
+            String fieldName;
+            switch (op) {
+                case CDCSourceConstants.CONNECT_RECORD_INSERT_OPERATION:
+                    //append row details after insert.
+                    rawDetails = (Struct) record.get(CDCSourceConstants.AFTER);
+                    fields = rawDetails.schema().fields();
+                    for (Field key : fields) {
+                        fieldName = key.name();
+                        detailsMap.put(fieldName, getValue(rawDetails.get(fieldName)));
+                    }
+                    break;
+                case CDCSourceConstants.CONNECT_RECORD_DELETE_OPERATION:
+                    //append row details before delete.
+                    rawDetails = (Struct) record.get(CDCSourceConstants.BEFORE);
+                    fields = rawDetails.schema().fields();
+                    for (Field key : fields) {
+                        fieldName = key.name();
+                        detailsMap.put(CDCSourceConstants.BEFORE_PREFIX + fieldName,
+                                getValue(rawDetails.get(fieldName)));
+                    }
+                    break;
+                case CDCSourceConstants.CONNECT_RECORD_UPDATE_OPERATION:
+                    //append row details before update.
+                    rawDetails = (Struct) record.get(CDCSourceConstants.BEFORE);
+                    fields = rawDetails.schema().fields();
+                    for (Field key : fields) {
+                        fieldName = key.name();
+                        detailsMap.put(CDCSourceConstants.BEFORE_PREFIX + fieldName,
+                                getValue(rawDetails.get(fieldName)));
+                    }
+                    //append row details after update.
+                    rawDetails = (Struct) record.get(CDCSourceConstants.AFTER);
+                    fields = rawDetails.schema().fields();
+                    for (Field key : fields) {
+                        fieldName = key.name();
+                        detailsMap.put(fieldName, getValue(rawDetails.get(fieldName)));
+                    }
+                    break;
+                default:
+                    log.info("Provided \"op\" value is not supported.");
```

Comment:
add the provided op value to the log as a context eg:- Provided value for op {value} is not supported

Reply:
Fixed